File: CPP/petshop.cpp
#include <iostream>
#include <string>

using namespace std;

// membuat kelas petshop
class Petshop {
private:
    int id;
    string namaProduk;
    string kategori;
    double harga;

public:
    // konstruktor default
    Petshop() {}

    // konstruktor dengan parameter
    Petshop(int id, string namaProduk, string kategori, double harga) {
        this->id = id;
        this->namaProduk = namaProduk;
        this->kategori = kategori;
        this->harga = harga;
    }

    // setter
    void setId(int id) { this->id = id; }
    void setNamaProduk(string namaProduk) { this->namaProduk = namaProduk; }
    void setKategori(string kategori) { this->kategori = kategori; }
    void setHarga(double harga) { this->harga = harga; }

    // getter
    int getId() { return id; }
    string getNamaProduk() { return namaProduk; }
    string getKategori() { return kategori; }
    double getHarga() { return harga; }

    // menampilkan data produk
    void tampilkan() {
        cout << "ID: " << id << endl;
        cout << "Nama Produk: " << namaProduk << endl;
        cout << "Kategori: " << kategori << endl;
        cout << "Harga: Rp" << harga << endl;
    }
};

File: Java/Petshop.java
package TP1DPBO2025.Java;

public class Petshop {
    private int id;
    private String namaProduk;
    private String kategori;
    private double harga;

    // konstruktor
    public Petshop(int id, String namaProduk, String kategori, double harga) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.kategori = kategori;
        this.harga = harga;
    }

    // setter
    public void setId(int id) { this.id = id; }
    public void setNamaProduk(String namaProduk) { this.namaProduk = namaProduk; }
    public void setKategori(String kategori) { this.kategori = kategori; }
    public void setHarga(double harga) { this.harga = harga; }

    // getter
    public int getId() { return id; }
    public String getNamaProduk() { return namaProduk; }
    public String getKategori() { return kategori; }
    public double getHarga() { return harga; }

    // menampilkan data produk
    public void tampilkan() {
        System.out.println("ID: " + id);
        System.out.println("Nama Produk: " + namaProduk);
        System.out.println("Kategori: " + kategori);
        System.out.println("Harga: Rp" + harga);
    }
}
